### DISCUSSION TITLE
Replacing Peak template path with an automated file locator

### DISCUSSION THREAD

**Discussion by kungphung** [studio1902/statamic-peak#229] [Ideas]
The Peak starter comes with antlers templates where you can add custom lines manually to show where the file is located in the 'view' folder.

In our projects we had automated this progress and we would like to have this in the Peak starter.

Peak example
![image](https://user-images.githubusercontent.com/50140858/175531119-663c7903-fafc-4b59-a526-32e1c96b0f1f.png)

Our example
![image](https://user-images.githubusercontent.com/50140858/175531154-cffd290e-b761-48f5-88b7-775879c42623.png)

In every new created antlers file the function will add the new line on top of the file with the path where the file is located therefore you don't have to add this line manually.

Also when your APP_DEBUG in your .env has the value "false" will hide/skip this function.
We have this in our production .env

With function
![image](https://user-images.githubusercontent.com/50140858/175532231-9a2e94ca-3626-486c-99c7-28d6905b127e.png)

Without function
![image](https://user-images.githubusercontent.com/50140858/175532619-d4f6c067-e308-45e5-a633-67c4bbf8e789.png)

If there are enough upvotes for this new feature maybe @robdekort can implement this in the starter.

**Comment by robdekort** [comment]
Hey Phung. Thanks. This came up before and I'm not a fan to be honest. It's a lot of conditionals just to get a few comments out that no end user will see. I don't really see the point. And for performance reasons I prefer the manual way. 

I assume `tid` is a custom tag? It still needs a manual parameter. When I compare it to the output it doesn't seem much time is saved.

Did you know when you run `php please peak:add-block` or `php please peak:add-set` it automatically generates the comments for you?

**Reply by robdekort** [reply]
Hey @kungphung. I didn't get a reply. Closing this for now. Hope all is well.

**Reply by robdekort** [reply]
Ah. I can't close an idea on GH discussions :-)

**Reply by kungphung** [reply]
Hey @robdekort ,

Sorry for my late reply.

The reason I want this in the standard is because I always use this in my projects as reference to find the correct file to change its contents when needed.
When there are more developers in a project it is a breeze to find the file quickly and you don't have to look for the right template when you have to change something.
Even for our support developers that has not build the project it will be easy to fix any bugs.
Note: This is for developers only and not for end users. Also this feature will disable any comment in the DOM for end users

I didn't knew the commands before and I will use them to create new blocks. But still I would like to have this in the standard.

If this is not possible or not neccesary for you I have to create something that can be installed with Peak.

**Reply by robdekort** [reply]
No worries. I don't think I understand what it is what you want.

1. The template comments are already in there. What are you proposing to add?
2. What is `{{ tid }}`
3. What is the the problem with shipping these comments to end users? It'll be gzipped and barely be any extra weight. Using conditionals will slow stuff down more (I assume).

**Reply by kungphung** [reply]
Hi @robdekort ,
(1.) My proposal is to add the tag TID that checks if environment is in development mode or production to show the (2.)TID(template in directory)
(3.) There is actually no problem showing the comments with the file path in the DOM, but I think it is cleaner not showing it. The function checks the environment first so I don't think things slow down (I assume :-) ).

**Reply by sjclark** [reply]
@kungphung 😄 whats the reason for this though - I'm assuming you've added this because your code editor can navigate straight to the file? (but can't from within the comment tag, in which case you might want to use one of the plugins that displays related files like https://marketplace.visualstudio.com/items?itemName=rioj7.html-related-links

Statamics code output always looks like trash anyways (❤️ ) so I don't really think the comments do anything bad - I personally prefer them when trying to do the odd front-end production diagnosis.

But it doesn't really feel worth the extra complexity to add the function/tag above - although it could be a candidate for some sort of independent addon that automatically adds links to related files?

If it's important to your internal workflow you could potentially just do a **find and replace** to achieve the same thing? Your function/tag is just replacing the start and end of the default view string in the comments with something else? 🤷‍♂️

**Reply by robdekort** [reply]
Hey Phung. 

Thanks! I'm very sorry to say but this is not something I feel Peak needs. Every time the tag gets called there's a conditional triggering and I simply don't see the benefit of shipping a cleaner source code. The rendered HTML is the only thing important for end users. Adding complexity with a tag feels a bit like over engineering to me. 

If you do add a tag for your use I'd def. consider fetching the path in PHP so you don't have to manually add that all the time. You could write a script that finds and replaces this automatically for you using a regex maybe.

No hard feelings I hope. Keep them ideas coming pls.

Best!

**Reply by kungphung** [reply]
Hi @robdekort and @sjclark,
No hard feelings indeed. I was just wondering what your thought was about this feature.
I allready have this in our project and I wil continue using this with your fabulous work with Peak :)
I will close this discussion.

Thnx for considering.

**Reply by robdekort** [reply]
Good to hear. TY.
